LayoutEditor-reorg

geowar1 opened the pull request:
Moving all LayoutEditor dialogs into LayoutEditorDialogs package(/folder)

geowar1 commented:
Ready to merge…

bobjacobsen commented:
Resolved conflict in layoutEditor/PackageTest

pabender commented:
@geowar1 this PR now has a conflict.  Not sure how it should be resolved. Please take a look at that.

pabender commented on java/src/jmri/jmrit/display/layoutEditor/LayoutEditorDialogs/Bundle.java:
```diff
@@ -0,0 +1,100 @@
+public class Bundle extends jmri.jmrit.display.layoutEditor.Bundle {
+
+    @CheckForNull
+    private static final String name = "jmri.jmrit.display.layoutEditor.LayoutEditorBundle"; // NOI18N
```
The bundle this file refers to needs to be in the package the file resides in.  If it isn’t. The name needs to be set to null.

The whole bundle system is designed so that if a key isn’t found in the local bundle you look to the parent bundle, but not following the rules bypasses the tree structure.

geowar1 replied:
Resource package file name set to null. Thanks for the review & catch.

geowar1 commented:
>>>AppVeyor was unable to build non-me
Ok, how do I fix this?

pabender commented:
There is still a merge conflict showing on github.  Fix that and hopefully the Appveyor build will complete.  ( I can’t see what the conflict is since the master branch is currently locked... )

bobjacobsen commented:
The conflicts are in `java/test/jmri/jmrit/display/layoutEditor/LayoutTrackEditorsTest.java`.  GitHub.com isn't giving an option to resolve on the web. From my own repo, I don't see a conflict, so I'm going to try a close/open.

bobjacobsen commented:
And here are the conflicts:
```
import jmri.Turnout;
<<<<<<< HEAD:java/test/jmri/jmrit/display/layoutEditor/LayoutEditorDialogs/LayoutTrackEditorsTest.java
import jmri.jmrit.display.layoutEditor.*;
=======
import jmri.TurnoutManager;
>>>>>>> master:java/test/jmri/jmrit/display/layoutEditor/LayoutTrackEditorsTest.java
import jmri.util.JUnitUtil;
```

```
/**
 * @author Paul Bender Copyright (C) 2017
<<<<<<< HEAD:java/test/jmri/jmrit/display/layoutEditor/LayoutEditorDialogs/LayoutTrackEditorsTest.java
 * @author George Warner Copyright: (c) 2019
=======
 * @author George Warner Copyright (C) 2019
>>>>>>> master:java/test/jmri/jmrit/display/layoutEditor/LayoutTrackEditorsTest.java
 */
public class LayoutTrackEditorsTest {
```

geowar1 commented:
I’m going to try a close/reopen as well…

geowar1 commented:
Still can’t see the conflicts… Looking at @bobjacobsen’s  post the head (top) versions are correct if there’s anyone that can resolve them.

geowar1 commented:
>@geowar1 Merge branch 'master' into LayoutEditor-reorg

230 files changed?!? I hate git…

bobjacobsen commented:
>230 files changed?!? I hate git…

Looks to me like Git did _exactly_ what you told it to do:

![image](https://user-images.githubusercontent.com/2774117/70855618-b9340f80-1e82-11ea-9add-3daf5d0e93bd.png)

When you merged `master` you picked up a bunch of changes.  

When you told git to _revert those changes_, you said "those are no good, and should be permanently removed from development".  Perhaps that's not what you wanted, and it's certainly not what the people who wrote all those changes wanted, but it's what you asked Git to do.

You may have intended to say "undo that merge", but that's not what reverting changes does.

There are several options for recovery, but it depends on what you really intended to do.  What was that?

geowar1 commented:
I did a "git merge origin/master"… (just like it says at <https://www.jmri.org/help/en/html/doc/Technical/GitFAQ.shtml#>) and git decided I'd changed 230 files… it works as expected about 9 time out of 10… but then…

Yeah, "reverting" shouldn't revert anything… that would make sense… git forbid that happening.

Maybe I should try one of these steps: <https://www.dev-metal.com/made-mess-git-heres-flowchart-guideline-fix/>. Nope; don't think so. I'll just make it worse.

I sure will be happy when the next version-control system replaces git… although it will probably be worse.

bobjacobsen commented:
> I did a "git merge origin/master"

Yes, you did.  That's marked in the log as b68dc50.  But the log _also_ shows that a `git revert` was done, marked as ef1b608.  If you didn't do that command, then something else did. Were you using the command line or something else?  Did you ^C out of a command or do something else unusual?

In any case, it would be good to _properly_ recover this.  What were you trying to do?

geowar1 commented:
I did the revert after the 'git merge origin/master" finished and GitHub decided I'd changed 230 files. Here's the % git reflog (if it helps):
ef1b608bf4 (HEAD -> LayoutEditor-reorg, origin/LayoutEditor-reorg) HEAD@{0}: checkout: moving from LayoutEditor-More-test-cases-for-LayoutTurntables-(code-coverage) to LayoutEditor-reorg
c72620c935 (origin/LayoutEditor-More-test-cases-for-LayoutTurntables-(code-coverage), LayoutEditor-More-test-cases-for-LayoutTurntables-(code-coverage)) HEAD@{1}: commit: Minor comment edit to restart the Travis CI build
9c2d9c49ce HEAD@{2}: checkout: moving from LayoutEditor-More-test-cases-for-LevelXing's to LayoutEditor-More-test-cases-for-LayoutTurntables-(code-coverage)
d273e69998 (origin/LayoutEditor-More-test-cases-for-LevelXing's, LayoutEditor-More-test-cases-for-LevelXing's) HEAD@{3}: checkout: moving from master to LayoutEditor-More-test-cases-for-LevelXing's
ddd4b97c87 (origin/master, origin/HEAD, master) HEAD@{4}: checkout: moving from LayoutEditor-More-test-cases-for-LayoutTurntables-(code-coverage) to master
9c2d9c49ce HEAD@{5}: checkout: moving from Test-cases-for-LayoutEditorChecks.java-(code-coverage-) to LayoutEditor-More-test-cases-for-LayoutTurntables-(code-coverage)
39806bb5b3 (origin/Test-cases-for-LayoutEditorChecks.java-(code-coverage-), Test-cases-for-LayoutEditorChecks.java-(code-coverage-)) HEAD@{6}: checkout: moving from Save-Panel-uses-panel-title-instead-of-load/last-saved-name to Test-cases-for-LayoutEditorChecks.java-(code-coverage-)
260974e47d (origin/Save-Panel-uses-panel-title-instead-of-load/last-saved-name, Save-Panel-uses-panel-title-instead-of-load/last-saved-name) HEAD@{7}: commit: Ok, I think that this reverts everything to the orginal behaviour
e70e5e1637 HEAD@{8}: checkout: moving from Test-cases-for-LayoutEditorChecks.java-(code-coverage-) to Save-Panel-uses-panel-title-instead-of-load/last-saved-name
39806bb5b3 (origin/Test-cases-for-LayoutEditorChecks.java-(code-coverage-), Test-cases-for-LayoutEditorChecks.java-(code-coverage-)) HEAD@{9}: commit: Minor comment edit to restart appveyor/pr
bbc8e82904 HEAD@{10}: checkout: moving from LayoutEditor-More-test-cases-for-LevelXing's to Test-cases-for-LayoutEditorChecks.java-(code-coverage-)
d273e69998 (origin/LayoutEditor-More-test-cases-for-LevelXing's, LayoutEditor-More-test-cases-for-LevelXing's) HEAD@{11}: commit: Done?
1f1e0d68fe HEAD@{12}: checkout: moving from LayoutEditor-More-test-cases-for-LayoutTurntables-(code-coverage) to LayoutEditor-More-test-cases-for-LevelXing's
9c2d9c49ce HEAD@{13}: commit: LayoutEditor-More-test-cases-for-LayoutTurntables (code coverage)
ddd4b97c87 (origin/master, origin/HEAD, master) HEAD@{14}: checkout: moving from master to LayoutEditor-More-test-cases-for-LayoutTurntables-(code-coverage)
ddd4b97c87 (origin/master, origin/HEAD, master) HEAD@{15}: checkout: moving from LayoutEditor-More-test-cases-for-LevelXing's to master
1f1e0d68fe HEAD@{16}: checkout: moving from Test-cases-for-LayoutEditorChecks.java-(code-coverage-) to LayoutEditor-More-test-cases-for-LevelXing's
bbc8e82904 HEAD@{17}: checkout: moving from master to Test-cases-for-LayoutEditorChecks.java-(code-coverage-)
ddd4b97c87 (origin/master, origin/HEAD, master) HEAD@{18}: checkout: moving from LayoutEditor-reorg to master
ef1b608bf4 (HEAD -> LayoutEditor-reorg, origin/LayoutEditor-reorg) HEAD@{19}: checkout: moving from master to LayoutEditor-reorg
ddd4b97c87 (origin/master, origin/HEAD, master) HEAD@{20}: checkout: moving from LayoutEditor-reorg to master
ef1b608bf4 (HEAD -> LayoutEditor-reorg, origin/LayoutEditor-reorg) HEAD@{21}: revert: Revert "Merge branch 'master' into LayoutEditor-reorg"
b68dc507fd HEAD@{22}: commit (merge): Merge branch 'master' into LayoutEditor-reorg
a349923f0f HEAD@{23}: checkout: moving from Test-cases-for-LayoutEditorChecks.java-(code-coverage-) to LayoutEditor-reorg
bbc8e82904 HEAD@{24}: merge master: Merge made by the 'recursive' strategy.
716d1cc999 HEAD@{25}: checkout: moving from LayoutEditor-More-test-cases-for-LevelXing's to Test-cases-for-LayoutEditorChecks.java-(code-coverage-)
1f1e0d68fe HEAD@{26}: merge master: Merge made by the 'recursive' strategy.
ee9d88b58d HEAD@{27}: checkout: moving from master to LayoutEditor-More-test-cases-for-LevelXing's
ddd4b97c87 (origin/master, origin/HEAD, master) HEAD@{28}: reset: moving to upstream/master
ddd4b97c87 (origin/master, origin/HEAD, master) HEAD@{29}: reset: moving to upstream/master
aeaf2879c4 HEAD@{30}: checkout: moving from LayoutEditor-More-test-cases-for-LevelXing's to master
ee9d88b58d HEAD@{31}: commit: Done?
c5bca2e134 HEAD@{32}: commit: More WIP
20f12706ba HEAD@{33}: commit: LayoutEditor:More test cases for LevelXing's
aeaf2879c4 HEAD@{34}: checkout: moving from master to LayoutEditor-More-test-cases-for-LevelXing's
aeaf2879c4 HEAD@{35}: reset: moving to upstream/master
aeaf2879c4 HEAD@{36}: reset: moving to upstream/master
30e38c157f HEAD@{37}: checkout: moving from Test-cases-for-LayoutEditorChecks.java-(code-coverage-) to master
716d1cc999 HEAD@{38}: commit: Added asserts to verify results of hard coded selectItem() values
4b2acffdc3 HEAD@{39}: commit: Added count checks for results menus
acb722a992 HEAD@{40}: commit: Minor comment edit to restart Travis-ci/pr... again
60db86f76c HEAD@{41}: commit: Minor comment edit to restart Travis-ci/pr
71faed8030 HEAD@{42}: merge origin/master: Merge made by the 'recursive' strategy.
4902939e2d HEAD@{43}: commit: DONE!
65680ad4f8 HEAD@{44}: commit: Saving more WIP
ef413afe06 HEAD@{45}: checkout: moving from Save-Panel-uses-panel-title-instead-of-load/last-saved-name to Test-cases-for-LayoutEditorChecks.java-(code-coverage-)
e70e5e1637 HEAD@{46}: commit: Fix for Travis-ci failure
597f76ab3f HEAD@{47}: commit: Fix for AppVeyor/pr failure
e0716a4b37 HEAD@{48}: commit: Minor comment edit to restart AppVeyor/pr
6191e51244 HEAD@{49}: commit: Bugfix:Save Panel uses panel title instead of load/last saved name
30e38c157f HEAD@{50}: checkout: moving from master to Save-Panel-uses-panel-title-instead-of-load/last-saved-name
30e38c157f HEAD@{51}: reset: moving to upstream/master
7794aa69cc HEAD@{52}: checkout: moving from LayoutEditor-fix-bug-introduced-by-#7682 to master
28bba4550f HEAD@{53}: checkout: moving from LayoutEditor-reorg to LayoutEditor-fix-bug-introduced-by-#7682
a349923f0f HEAD@{54}: commit: Minor comment edit to restart AppVeyor/pr
8daf998387 HEAD@{55}: checkout: moving from Test-cases-for-LayoutEditorChecks.java-(code-coverage-) to LayoutEditor-reorg
ef413afe06 HEAD@{56}: pull --progress origin: Merge made by the 'recursive' strategy.
d944f27eda HEAD@{57}: checkout: moving from LayoutEditor-fix-bug-introduced-by-#7682 to Test-cases-for-LayoutEditorChecks.java-(code-coverage-)
28bba4550f HEAD@{58}: commit: Update panel.js
35c3ade29e HEAD@{59}: commit: LayoutEditor-fix bug introduced by #7682
7794aa69cc HEAD@{60}: checkout: moving from master to LayoutEditor-fix-bug-introduced-by-#7682
7794aa69cc HEAD@{61}: reset: moving to upstream/master
750a708bbf HEAD@{62}: checkout: moving from LayoutEditor-LayoutTrack-Refactor to master
3037812376 HEAD@{63}: checkout: moving from master to LayoutEditor-LayoutTrack-Refactor
750a708bbf HEAD@{64}: checkout: moving from LayoutEditor-Explore-Alternative-Slip-Coloring to master
5d87ab8ce5 HEAD@{65}: checkout: moving from master to LayoutEditor-Explore-Alternative-Slip-Coloring
750a708bbf HEAD@{66}: checkout: moving from LayoutEditor-Code-Coverage-and-Test-Cases to master
ad5a6303f7 HEAD@{67}: checkout: moving from LayoutEditor-reorg to LayoutEditor-Code-Coverage-and-Test-Cases
8daf998387 HEAD@{68}: commit: Set resource package file name to null
561465a40e HEAD@{69}: commit: Moved conflicting files into jmri.jmrit.display.layoutEditor.LayoutEditorDialogs (sub-)package.
b0b638387d HEAD@{70}: checkout: moving from LayoutEditor-Explore-Alternative-Slip-Coloring to LayoutEditor-reorg
5d87ab8ce5 HEAD@{71}: checkout: moving from master to LayoutEditor-Explore-Alternative-Slip-Coloring
750a708bbf HEAD@{72}: checkout: moving from LayoutEditor-Add-GUI-to-set-LayoutSlip-blocks-B,-C-and-D to master
ef3815cf5c HEAD@{73}: checkout: moving from LayoutEditor-Explore-Alternative-Slip-Coloring to LayoutEditor-Add-GUI-to-set-LayoutSlip-blocks-B,-C-and-D
5d87ab8ce5 HEAD@{74}: commit: minor clean up and reformatting…
6472edd637 HEAD@{75}: commit: ...try, try again...
5818e01a64 HEAD@{76}: commit: Non-route leg colors match adjacent track segment block colors.
6708000a98 HEAD@{77}: commit: Alternative way to draw slip colors (based on adjacent tracksegments)
750a708bbf HEAD@{78}: checkout: moving from master to LayoutEditor-Explore-Alternative-Slip-Coloring
750a708bbf HEAD@{79}: reset: moving to upstream/master
10d5cc8a69 HEAD@{80}: checkout: moving from LayoutEditor-Add-GUI-to-set-LayoutSlip-blocks-B,-C-and-D to master
ef3815cf5c HEAD@{81}: checkout: moving from Test-cases-for-LayoutEditorChecks.java-(code-coverage-) to LayoutEditor-Add-GUI-to-set-LayoutSlip-blocks-B,-C-and-D
d944f27eda HEAD@{82}: commit: Save WIP
b542e94ced HEAD@{83}: checkout: moving from master to Test-cases-for-LayoutEditorChecks.java-(code-coverage-)
10d5cc8a69 HEAD@{84}: checkout: moving from LayoutEditor-webserver-layoutslip-fixes to master
d0b7c88bf9 HEAD@{85}: commit: LayoutEditor webserver layoutslip fixes
10d5cc8a69 HEAD@{86}: checkout: moving from master to LayoutEditor-webserver-layoutslip-fixes
10d5cc8a69 HEAD@{87}: reset: moving to HEAD
10d5cc8a69 HEAD@{88}: checkout: moving from Test-cases-for-LayoutEditorChecks.java-(code-coverage-) to master
b542e94ced HEAD@{89}: commit: save WIP
3c8336e423 HEAD@{90}: commit: Added testBundleStrings
87d07e5027 HEAD@{91}: commit: Added missing Assume.assumeFalse(GraphicsEnvironment.isHeadless());
b13f494044 HEAD@{92}: checkout: moving from master to Test-cases-for-LayoutEditorChecks.java-(code-coverage-)
10d5cc8a69 HEAD@{93}: checkout: moving from LayoutEditor-Add-GUI-to-set-LayoutSlip-blocks-B,-C-and-D to master
ef3815cf5c HEAD@{94}: checkout: moving from master to LayoutEditor-Add-GUI-to-set-LayoutSlip-blocks-B,-C-and-D
10d5cc8a69 HEAD@{95}: reset: moving to upstream/master
bfac31aea1 HEAD@{96}: checkout: moving from LayoutEditor-Add-GUI-to-set-LayoutSlip-blocks-B,-C-and-D to master
ef3815cf5c HEAD@{97}: commit: LayoutEditor Add GUI to set LayoutSlip blocks B, C and D
bfac31aea1 HEAD@{98}: Branch: renamed refs/heads/LayoutEditor-No-GUI-to-set-LayoutSlip-blocks-B,-C-and-D to refs/heads/LayoutEditor-Add-GUI-to-set-LayoutSlip-blocks-B,-C-and-D
bfac31aea1 HEAD@{100}: checkout: moving from master to LayoutEditor-No-GUI-to-set-LayoutSlip-blocks-B,-C-and-D
bfac31aea1 HEAD@{101}: checkout: moving from LayoutEditor-webserver-displays-XOvers,-Crossings-and-Slips-incorrectly to master
7ff6df141d HEAD@{102}: checkout: moving from Test-cases-for-LayoutEditorChecks.java-(code-coverage-) to LayoutEditor-webserver-displays-XOvers,-Crossings-and-Slips-incorrectly
b13f494044 HEAD@{103}: commit: trying pushMenuNoBlock
72077efad5 HEAD@{104}: commit: added Assert.assertNotNull's for Travis CI #19297.2 failed
0a9b59612f HEAD@{105}: checkout: moving from LayoutEditor-reorg to Test-cases-for-LayoutEditorChecks.java-(code-coverage-)
2752fa5e3a HEAD@{106}: commit: Minor comment edit to restart AppVeyor
b5d1f3a934 HEAD@{107}: checkout: moving from Test-cases-for-LayoutEditorChecks.java-(code-coverage-) to LayoutEditor-reorg
0a9b59612f HEAD@{108}: commit: Add Bundle.getMessage string checks for Travis-ci/pr #19288.2 failed
dd7eb71b7d HEAD@{109}: checkout: moving from LayoutEditor-reorg to Test-cases-for-LayoutEditorChecks.java-(code-coverage-)
b5d1f3a934 HEAD@{110}: commit: Added public int hashCode() (for Travis-ci/pr #19387 failed)
c5294690dd HEAD@{111}: checkout: moving from Test-cases-for-LayoutEditorChecks.java-(code-coverage-) to LayoutEditor-reorg
dd7eb71b7d HEAD@{112}: commit: fixes for #19284.3 failed
6105652a1b HEAD@{113}: checkout: moving from LayoutEditor-reorg to Test-cases-for-LayoutEditorChecks.java-(code-coverage-)
c5294690dd HEAD@{114}: commit: Fix for The Travis CI build
67fed895c2 HEAD@{115}: commit: Fix TurnoutState.equals method (for Travis-ci/pr #19283 failed)
af2bd81821 HEAD@{116}: checkout: moving from Test-cases-for-LayoutEditorChecks.java-(code-coverage-) to LayoutEditor-reorg
6105652a1b HEAD@{117}: commit: More LayoutEditorChecksTest work
3840d3d4d8 HEAD@{118}: checkout: moving from LayoutEditor-Code-Coverage-and-Test-Cases to Test-cases-for-LayoutEditorChecks.java-(code-coverage-)
ad5a6303f7 HEAD@{119}: checkout: moving from LayoutEditor-reorg to LayoutEditor-Code-Coverage-and-Test-Cases
af2bd81821 HEAD@{120}: commit: Fix javadoc
11d83b864a HEAD@{121}: checkout: moving from LayoutEditor-webserver-displays-XOvers,-Crossings-and-Slips-incorrectly to LayoutEditor-reorg
7ff6df141d HEAD@{122}: commit: LayoutEditor webserver displays XOvers, Crossings and Slips incorrectly
bfac31aea1 HEAD@{123}: checkout: moving from master to LayoutEditor-webserver-displays-XOvers,-Crossings-and-Slips-incorrectly
bfac31aea1 HEAD@{124}: reset: moving to upstream/master
bfac31aea1 HEAD@{125}: checkout: moving from LayoutEditor-Code-Coverage-and-Test-Cases to master
ad5a6303f7 HEAD@{126}: commit: minor comment edit to restart Travis/ci
af665e1858 HEAD@{127}: checkout: moving from Test-cases-for-LayoutEditorChecks.java-(code-coverage-) to LayoutEditor-Code-Coverage-and-Test-Cases
3840d3d4d8 HEAD@{128}: commit: Test cases for LayoutEditorChecks.java (code coverage )
bfac31aea1 HEAD@{129}: checkout: moving from master to Test-cases-for-LayoutEditorChecks.java-(code-coverage-)
bfac31aea1 HEAD@{130}: checkout: moving from LayoutEditor-reorg to master
11d83b864a HEAD@{131}: checkout: moving from LayoutEditor-Code-Coverage-and-Test-Cases to LayoutEditor-reorg
af665e1858 HEAD@{132}: commit: Delete test file
7d7d5cc807 HEAD@{133}: commit: Minor comment edit to restart Travis/ci
be05a558bf HEAD@{134}: checkout: moving from LayoutEditor-reorg to LayoutEditor-Code-Coverage-and-Test-Cases
11d83b864a HEAD@{135}: commit: LayoutEditor-reorg
bfac31aea1 HEAD@{136}: checkout: moving from master to LayoutEditor-reorg
bfac31aea1 HEAD@{137}: reset: moving to upstream/master
c7a9ae6779 HEAD@{138}: checkout: moving from LayoutEditor-reorg to master
c7a9ae6779 HEAD@{139}: checkout: moving from master to LayoutEditor-reorg
c7a9ae6779 HEAD@{140}: checkout: moving from LayoutEditor-Code-Coverage-and-Test-Cases to master
be05a558bf HEAD@{141}: commit: More code cleanup
fe8c1fe4a9 HEAD@{142}: commit: Minor comment edit to restart Travis/ci
67851e5e7a HEAD@{143}: commit: minor comment edit to restart Appveyor
a4993a9971 HEAD@{144}: commit: Cut & paste error... typo...
4f293fbd66 HEAD@{145}: commit: Added testSetupSSL
8a45d06615 HEAD@{146}: commit: Break testToolBarPositionOptions into four parts
265ca1180e HEAD@{147}: commit: clean up after ourselves
cece4ab7f8 HEAD@{148}: commit: Minor comment edit to restart Appveyor/ci
eaf1f96747 HEAD@{149}: commit: Finish(?) testSetSignalsAtTurnoutWithDone work...
3aafef79d2 HEAD@{150}: commit: More testSetSignalsAtTurnoutWithDone work
33a30a90c6 HEAD@{151}: commit: Finish LayoutEditorToolsTest.testSetSignalsAtTurnoutWithDone(...)
d7616a7aa5 HEAD@{152}: commit: restore broken conditionals
9c6b6442f8 HEAD@{153}: checkout: moving from master to LayoutEditor-Code-Coverage-and-Test-Cases
c7a9ae6779 HEAD@{154}: checkout: moving from BugFix-LayoutEditorToolsTest.testSetSignalsAtLevelXing-FromMenu]-@Ignore to master
41bc373cee HEAD@{155}: commit: bugfix for #19199.3 failed
a5a5db93ee HEAD@{156}: checkout: moving from LayoutEditor-Code-Coverage-and-Test-Cases to BugFix-LayoutEditorToolsTest.testSetSignalsAtLevelXing-FromMenu]-@Ignore
9c6b6442f8 HEAD@{157}: commit: bugfix for #19199.3 failed
35ce924cac HEAD@{158}: commit: bugfix and tweek comments
159f21b98d HEAD@{159}: checkout: moving from Bugfix-LayoutEditorTools-Signal-heads-created-withou-icons to LayoutEditor-Code-Coverage-and-Test-Cases
86a50feb81 HEAD@{160}: checkout: moving from BugFix-LayoutEditorToolsTest.testSetSignalsAtLevelXing-FromMenu]-@Ignore to Bugfix-LayoutEditorTools-Signal-heads-created-withou-icons
a5a5db93ee HEAD@{161}: commit: Bugfix:LayoutEditorToolsTest
04ac8e2c7c HEAD@{162}: commit: BugFix: LayoutEditorToolsTest.testSetSignalsAtLevelXing[FromMenu] @Ignore
c7a9ae6779 HEAD@{163}: checkout: moving from master to BugFix-LayoutEditorToolsTest.testSetSignalsAtLevelXing-FromMenu]-@Ignore
c7a9ae6779 HEAD@{164}: checkout: moving from Bugfix-LayoutEditorTools-Signal-heads-created-withou-icons to master
86a50feb81 HEAD@{165}: checkout: moving from master to Bugfix-LayoutEditorTools-Signal-heads-created-withou-icons
c7a9ae6779 HEAD@{166}: reset: moving to HEAD
c7a9ae6779 HEAD@{167}: checkout: moving from Bugfix-LayoutEditorTools-Signal-heads-created-withou-icons to master
86a50feb81 HEAD@{168}: checkout: moving from master to Bugfix-LayoutEditorTools-Signal-heads-created-withou-icons
c7a9ae6779 HEAD@{169}: checkout: moving from LayoutEditor-Code-Coverage-and-Test-Cases to master
159f21b98d HEAD@{170}: commit: Reenable @Tests
d23c30a17d HEAD@{171}: checkout: moving from Bugfix-LayoutEditorTools-Signal-heads-created-withou-icons to LayoutEditor-Code-Coverage-and-Test-Cases
86a50feb81 HEAD@{172}: commit: Update LayoutEditorTools.java
c7a9ae6779 HEAD@{173}: checkout: moving from master to Bugfix-LayoutEditorTools-Signal-heads-created-withou-icons
c7a9ae6779 HEAD@{174}: reset: moving to HEAD
c7a9ae6779 HEAD@{175}: reset: moving to upstream/master
c7a9ae6779 HEAD@{176}: reset: moving to upstream/master
1b1a8c1be3 HEAD@{177}: checkout: moving from LayoutEditor-Code-Coverage-and-Test-Cases to master
d23c30a17d HEAD@{178}: commit: LayoutEditor Code Coverage and Test Cases
1b1a8c1be3 HEAD@{179}: checkout: moving from master to LayoutEditor-Code-Coverage-and-Test-Cases
1b1a8c1be3 HEAD@{180}: reset: moving to HEAD
1b1a8c1be3 HEAD@{181}: reset: moving to upstream/master
49102131d3 HEAD@{182}: checkout: moving from Fix-LayoutEditor-code-climate-issues to master
085f7088ab HEAD@{183}: commit: Finished LayoutTrackEditorsTest
2e2915f5d4 HEAD@{184}: commit: travis-ci/pr fixes
05dc73bfdb HEAD@{185}: commit: More LayoutTrackEditors test code work
527c39e8dd HEAD@{186}: commit: Working on test code
58e3221fe1 HEAD@{187}: commit: Removed static on @Before/!After methods
b5fa00ccd9 HEAD@{188}: commit: Removed @BeforeClass/@AfterClass
a6ab6c05fb HEAD@{189}: commit: Appveyor faile: LinkedWarrantTest
e708e7d449 HEAD@{190}: commit: Travis-ci/pr fixes
71b2416031 HEAD@{191}: commit: Removed @BeforeClass/@AfterClass from LayoutEditorTest
b2b0c451a6 HEAD@{192}: commit: Made all the changes @pbender suggested
840c1c2873 HEAD@{193}: commit: Dead code strip & fix javadocs
e82806a216 HEAD@{194}: commit: Finished test cases for LayoutEditorDialogs!
d5bfe1d417 HEAD@{195}: commit: @geowar1 Edit comment to restart Appveyor/pr
4e7a5409b8 HEAD@{196}: commit: Edit comment to restart Appveyor/pr
5ec4704ef6 HEAD@{197}: commit: Update javadocs
7622d7df72 HEAD@{198}: commit: Added LayoutEditorDialogsTest!
d443cc8aa6 HEAD@{199}: commit: more code climate
92d145d9e7 HEAD@{200}: commit: Move LayoutEditor dialogs to LayoutEditorDialogs.java
a3d274cabf HEAD@{201}: commit: spotbugs fixes for travis-ci/pr
935ff0a283 HEAD@{202}: commit: Update QuickPromptUtil.java
d8f54be1b1 HEAD@{203}: commit: BugFix for QuickPromptUtil.promptForInteger()
8871609028 HEAD@{204}: commit: Add @Nonnull
012e4c6cda HEAD@{205}: commit: Initial commit
49102131d3 HEAD@{206}: checkout: moving from master to Fix-LayoutEditor-code-climate-issues
49102131d3 HEAD@{207}: reset: moving to HEAD
49102131d3 HEAD@{208}: reset: moving to upstream/master
157d753feb HEAD@{209}: checkout: moving from WebServer-doesn't-show-Occupied-and-Alternative-track-color-in-Panel-in-web-browser to master
f43ad23402 HEAD@{210}: commit: edit comment to restart appveyor/pr
72479c7e69 HEAD@{211}: commit: Comment out debug logging statement
e2cdb667a8 HEAD@{212}: commit: WebServer doesn't show Occupied and Alternative track color in Panel in web browser
157d753feb HEAD@{213}: checkout: moving from master to WebServer-doesn't-show-Occupied-and-Alternative-track-color-in-Panel-in-web-browser
157d753feb HEAD@{214}: reset: moving to upstream/master
16f1aabfed HEAD@{215}: reset: moving to HEAD
16f1aabfed HEAD@{216}: checkout: moving from LayoutShape-bugfixes to master
0d20716be4 HEAD@{217}: commit: edit comments to restart appveyer
f787ed77c5 HEAD@{218}: commit: Disable non-production (debug) log statements
493ba95edf HEAD@{219}: commit: Bugfix for #7454
041116f82f HEAD@{220}: commit: Add Layout Shapes to client-side web panels
16f1aabfed HEAD@{221}: checkout: moving from master to LayoutShape-bugfixes
16f1aabfed HEAD@{222}: checkout: moving from Add-LE-Shapes-to-client-side-web-panels to master
3ce7de1568 HEAD@{223}: checkout: moving from master to Add-LE-Shapes-to-client-side-web-panels
16f1aabfed HEAD@{224}: reset: moving to upstream/master
3ce7de1568 HEAD@{225}: checkout: moving from Add-LE-Shapes-to-client-side-web-panels to master
3ce7de1568 HEAD@{226}: reset: moving to HEAD
3ce7de1568 HEAD@{227}: reset: moving to HEAD
3ce7de1568 HEAD@{228}: checkout: moving from master to Add-LE-Shapes-to-client-side-web-panels
3ce7de1568 HEAD@{229}: checkout: moving from Add-LE-Decorations-to-client-side-web-panels to master
334978e177 HEAD@{230}: checkout: moving from master to Add-LE-Decorations-to-client-side-web-panels
3ce7de1568 HEAD@{231}: checkout: moving from Add-LE-Decorations-to-client-side-web-panels to master
334978e177 HEAD@{232}: merge JMRI/master: Merge made by the 'recursive' strategy.
2b281819eb HEAD@{233}: commit: WarrantTest… again…
1500957883 HEAD@{234}: commit: More LinkedWarrantTest fails…
cf1d505a63 HEAD@{235}: merge JMRI/master: Merge made by the 'recursive' strategy.
883fc9bf72 HEAD@{236}: commit: comment edit to force appveyor/pr restart
64ae28fccf HEAD@{237}: commit: Edit comment to restart appveyor/pr
c9ab7b6f99 HEAD@{238}: commit: edit comment to restart appveyor/pr
294a3e621c HEAD@{239}: commit: Update TrackSegmentTest.java
12f0d82c74 HEAD@{240}: commit: edited comment to restart appveyor/pr
25899b30e4 HEAD@{241}: commit: added comment to restart travis.ci
4386967528 HEAD@{242}: commit: add missing headless conditionals
c8aaa79894 HEAD@{243}: commit: Wrote a bunch of test cases for TrackSegment and MathUtil
44de22a5d7 HEAD@{244}: checkout: moving from master to Add-LE-Decorations-to-client-side-web-panels
3ce7de1568 HEAD@{245}: checkout: moving from Improve-code-coverage-for-LayoutEditor-package to master
3ce7de1568 HEAD@{246}: checkout: moving from master to Improve-code-coverage-for-LayoutEditor-package
3ce7de1568 HEAD@{247}: checkout: moving from Add-LE-Decorations-to-client-side-web-panels to master
44de22a5d7 HEAD@{248}: reset: moving to HEAD
44de22a5d7 HEAD@{249}: checkout: moving from master to Add-LE-Decorations-to-client-side-web-panels
3ce7de1568 HEAD@{250}: checkout: moving from Add-LE-Decorations-to-client-side-web-panels to master
44de22a5d7 HEAD@{251}: commit: Called it! Removing the WarrantManagerXml line now…
6db114e8c3 HEAD@{252}: commit: Removing OBlockManagerXml line from Decorations.xml:36
9e409dd2cc HEAD@{253}: commit: Remove signalmasts and signalgroups because of failing ci
5a27d2474a HEAD@{254}: commit: Touch a file to restart travis-ci
ef56328a07 HEAD@{255}: commit: Touching Decorations.xml to force CI restart
d51257eec4 HEAD@{256}: commit: forgot "public" on intersect…
1b22f138f7 HEAD@{257}: commit: Add TODO…
94e29d0879 HEAD@{258}: commit: Added intersect method to MathUtil
584fa12955 HEAD@{259}: checkout: moving from master to Add-LE-Decorations-to-client-side-web-panels
3ce7de1568 HEAD@{260}: checkout: moving from Add-code-to-calculate-projected-intersection-of-two-tracks to master
3ce7de1568 HEAD@{261}: reset: moving to HEAD
3ce7de1568 HEAD@{262}: checkout: moving from master to Add-code-to-calculate-projected-intersection-of-two-tracks
3ce7de1568 HEAD@{263}: checkout: moving from Add-LE-Decorations-to-client-side-web-panels to master
584fa12955 HEAD@{264}: commit: Minor refactoring
ada842a26e HEAD@{265}: commit: Minor optimization, bugfix
b9b28e0383 HEAD@{266}: commit: More codeclimate...
272c76e89e HEAD@{267}: commit: dead code strip & more code climate
f970596d68 HEAD@{268}: commit: More codeclimate...
bc37e9bf0c HEAD@{269}: commit: More codeclimate work...
a2edc73d3c HEAD@{270}: commit: More codeclimate work...
2c4539c4d1 HEAD@{271}: commit: Use $gCtx.save, .translate, .rotate, restore instead of $point math.
9ce8ede673 HEAD@{272}: commit: Add tunnel decorations class
885aee31a6 HEAD@{273}: commit: packing code for codeclimate
02d57714c3 HEAD@{274}: commit: Update LE Decorations test panels
1da2229c65 HEAD@{275}: commit: Add bumper decorations class
04b4cb9df6 HEAD@{276}: commit: WIP: move arrow and bumper decorations into classes
10c7a39576 HEAD@{277}: merge JMRI/master: Merge made by the 'recursive' strategy.
941a750ab1 HEAD@{278}: commit: Initial (re-)checkin
3ce7de1568 HEAD@{279}: checkout: moving from master to Add-LE-Decorations-to-client-side-web-panels
3ce7de1568 HEAD@{280}: reset: moving to HEAD
3ce7de1568 HEAD@{281}: reset: moving to upstream/master
30de10dacb HEAD@{282}: checkout: moving from Add-Layout-Editor-Decorations-to-client-side-web-panels to master
b37c1fd972 HEAD@{283}: checkout: moving from master to Add-Layout-Editor-Decorations-to-client-side-web-panels
30de10dacb HEAD@{284}: checkout: moving from Add-Layout-Editor-Decorations-to-client-side-web-panels to master
b37c1fd972 HEAD@{285}: pull --progress origin: Merge made by the 'recursive' strategy.
3ce7de1568 HEAD@{286}: reset: moving to upstream/master
712a4cf997 HEAD@{287}: commit: New decoration test cases!
a93ef60acd HEAD@{288}: commit: Bugfix: accessor didn't save new value.
cc4eceac06 HEAD@{289}: commit: Update Decorations.xml
b50cf12acf HEAD@{290}: commit: LoadAndStoreTest didn't like window size
cded67fb3f HEAD@{291}: commit: Added Decorations.XML to loadref
e07ac7411c HEAD@{292}: commit: Recover from git insanity
30de10dacb HEAD@{293}: reset: moving to master
5e95f2f52b HEAD@{294}: commit: More code decomplexification
ef111a1d34 HEAD@{295}: commit: remove some code complexity
ee159fea09 HEAD@{296}: commit: Add Decorations layout to WebPanel.feature (for testing)
98197a85c0 HEAD@{297}: pull --progress origin: Merge made by the 'recursive' strategy.
4df4423fb5 HEAD@{298}: reset: moving to upstream/master
909320f8c1 HEAD@{299}: merge origin/master: Merge made by the 'recursive' strategy.
725c9a2827 HEAD@{300}: commit: fix failing testDefaultGetSetBumperLineWidth test
dc1c2827e4 HEAD@{301}: commit: Code tweaks and bug fixes
1c399d55f8 HEAD@{302}: commit: Finished tunnel drawing code!
bc7f1bcaeb HEAD@{303}: commit: Added $drawBumper function
b35cefc241 HEAD@{304}: commit: Minor refactoring for codeclimate
92ed289265 HEAD@{305}: commit: Fix start/stopAngle calculations and $drawEllipse parameters
6437edd737 HEAD@{306}: checkout: moving from master to Add-Layout-Editor-Decorations-to-client-side-web-panels
30de10dacb HEAD@{307}: reset: moving to upstream/master
30de10dacb HEAD@{308}: reset: moving to upstream/master
156c190607 HEAD@{309}: checkout: moving from Add-Layout-Editor-Decorations-to-client-side-web-panels to master
6437edd737 HEAD@{310}: commit: Arrow decorations finished!
3227a7c02e HEAD@{311}: commit: WIP:working on start/stopAngle and arrow drawing code
a99b05764e HEAD@{312}: commit: Add displacement to Bezier functions
31239d143d HEAD@{313}: commit: Fix tabs -> spaces
08232a6929 HEAD@{314}: commit: Add Layout Editor Decorations to client-side web panels
156c190607 HEAD@{315}: checkout: moving from master to Add-Layout-Editor-Decorations-to-client-side-web-panels
156c190607 HEAD@{316}: reset: moving to upstream/master
156c190607 HEAD@{317}: reset: moving to HEAD
156c190607 HEAD@{318}: reset: moving to upstream/master
156c190607 HEAD@{319}: reset: moving to upstream/master
5d9b7ea040 HEAD@{320}: reset: moving to upstream/master
13ce6bf2ea HEAD@{321}: reset: moving to upstream/master
13ce6bf2ea HEAD@{322}: reset: moving to upstream/master
cb40eb8930 HEAD@{323}: checkout: moving from Layout-Editor-Turnout-Color-Change to master
5fed8f5ff4 HEAD@{324}: reset: moving to upstream/master
5fed8f5ff4 HEAD@{325}: reset: moving to upstream/master

bobjacobsen commented:
The prior commit on your LayoutEditor-reorg branch was a349923.  When I check that out, then merge ddd4b97  (the SHA you merged from master) I get:
```
% git merge ddd4b97
Removing lib/json-schema-validator-1.0.10.jar
Auto-merging java/test/jmri/jmrit/display/layoutEditor/LayoutEditorDialogs/LayoutTrackEditorsTest.java
CONFLICT (content): Merge conflict in java/test/jmri/jmrit/display/layoutEditor/LayoutEditorDialogs/LayoutTrackEditorsTest.java
Auto-merging java/src/jmri/jmrit/display/layoutEditor/LevelXing.java
Auto-merging java/src/jmri/jmrit/display/layoutEditor/LayoutTurnout.java
Auto-merging java/src/jmri/jmrit/display/layoutEditor/LayoutSlip.java
Auto-merging java/src/jmri/jmrit/display/layoutEditor/LayoutEditor.java
Auto-merging java/src/jmri/jmrit/display/layoutEditor/LayoutBlock.java
Automatic merge failed; fix conflicts and then commit the result.
```
The conflicts are trivial:  Extra imports, and removing a colon from the copyright light.  After fixing those, I get this commit log:
[COMMIT_EDITMSG.txt](https://github.com/JMRI/JMRI/files/3964543/COMMIT_EDITMSG.txt)
which shows 185 "modified" lines because of all the files that have been merged on master since the LayoutEditor-reorg branch was merged to it. 

Does that look like what happened when you did the merge?

After that merge commit, `git diff master...` shows 26 files changed, which is about right for this branch.

I pushed that to GitHub and proposed a PR.  It also shows 26 files, not 230.

So you must have done something else.

The only way I've found to have GitHub see 230 changed files is to push and check _after_ reverting.  Are you sure that's not what you did?

bobjacobsen commented:
Pushing a branch at the b68dc507fdcc55379c01e1b3cbb6f33dce0a5a8c SHA (the merge, before the revert) shows 26 files changed.  So, if that's the content you want, you should reset the branch to that and push it again. (NOTE:  This is not a "revert"; "revert" means something completely different, and most people should never use "revert")

```
git checkout b68dc507fdcc55379c01e1b3cbb6f33dce0a5a8c   
git branch -f LayoutEditor-reorg HEAD  
git checkout LayoutEditor-reorg
git push -f
```

Alternately, you can create a new branch and make a new PR from it:
```
git checkout b68dc507fdcc55379c01e1b3cbb6f33dce0a5a8c  
git checkout -b LayoutEditor-reorg-replacement
git push --set-upstream origin LayoutEditor-reorg-replacement
```

geowar1 commented:
Ok, fingers crossed…

geowar1 commented:
[ERROR] /home/travis/build/JMRI/JMRI/java/test/jmri/jmrit/display/layoutEditor/LayoutEditorDialogs/LayoutTrackEditorsTest.java:[804,47] cannot find symbol class TurnoutManager
I'll try jmri.TurnoutManager.class…

bobjacobsen commented on java/test/jmri/jmrit/display/layoutEditor/LayoutEditorDialogs/LayoutTrackEditorsTest.java:
```diff
@@ -801,11 +795,11 @@ public void testEditTurntableErrors() {
 
     public void createTurnouts() {
-        turnout0 = InstanceManager.getDefault(TurnoutManager.class).provideTurnout("IT101");
+        turnout0 = InstanceManager.getDefault(jmri.TurnoutManager.class).provideTurnout("IT101");
```
Once you've added the `import jmri.*`, the `jmri.` prefix isn't needed here.  Before the `import jmri.*` was added, `jmri.TurnoutManager` wasn't being imported.

geowar1 replied:
I was getting an error (on Travis CI) that it couldn’t find the TurnoutManager.class… I added the jmri prefix specifically to fix that.

geowar1 replied:
I’ll check the import… Thanks.

bobjacobsen commented:
close and reopen to reset the CI branch

geowar1 commented:
This is ready to merge…

bobjacobsen commented:
@pabender You've still got this marked as "change requested". It looks to me like that was resolved, but I'm not sure.  Is this OK to merge to development?

geowar1 commented:
Yes (resolved & merge) please!

pabender commented:
Merged.  Thanks @geowar1 and @bobjacobsen
